swift/apple/FirezoneKit/Sources/FirezoneKit/Views/UpdateNotification.swift

```diff
@@ -46,29 +46,46 @@ class UpdateChecker {
         let task = URLSession.shared.dataTask(with: versionCheckUrl) { [weak self] data, response, error in
           guard let self = self else { return }
 
-          if let error = error {
-            Log.error(error)
-            return
-          }
+          do {
```

Comment:
Why do you need the `do` block? Wouldn't it be cleaner to do the case handling within this `if` at the top? Then you avoid a level of indentation for the rest of it.

Also, `throw` and immediate `catch` seems like a code-smell.

Reply:
There are two `throws` in this body. I was applying [the same convention we used](https://github.com/firezone/firezone/pull/7752) for PacketTunnelProvider but taking a closer look, we don't `try` anything here -- our only errors are our own.

Updated.